src/test/kotlin/com/beust/klaxon/TestRFC7159.kt

```diff
@@ -0,0 +1,134 @@
+package com.beust.klaxon;
+
+import org.testng.annotations.Test
+import java.io.ByteArrayInputStream
+import kotlin.reflect
+import kotlin.reflect.KClass
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+class TestRFC7159 {
+
+    fun parse(jsonData: String) : Any? {
+        return Parser().parse(ByteArrayInputStream(jsonData.toByteArray(Charsets.UTF_8)))
+    }
+
+    fun jsonArrayEquals(testData: String, expectedData: String) {
+        val j = parse(testData)
+        if (j is JsonArray<*>) {
+            assertEquals(expectedData, j.toJsonString())
+        } else {
+            fail("not an array")
+        }
+    }
+
+    fun jsonObjectEquals(testData: String, expectedData: String) {
+        val j = parse(testData)
+        if (j is JsonObject) {
+            assertEquals(expectedData, j.toJsonString())
+        } else {
+            fail("not an object")
+        }
+    }
+
+    Test
```

Comment:
Might want to put a single `@Test` at the top of the class instead of repeating the annotation everywhere.